datasets/mode.py

```diff
@@ -0,0 +1,10 @@
+from enum import Flag, auto
+
+
+class Mode(Flag):
+    """
+    Used to represent data access modes.
+    """
+
+    READ = auto()
+    WRITE = auto()
```

Comment:
When a dataset is in `WRITE` mode, can it also be read?

Reply:
yes, if you can write you can read.  Is that confusing?

Reply:
A little, normally if that's the case I'd expect it to be called `READ_WRITE` as typically if you just have `write` permission you cannot `read` but maybe I'm just overly stuck in my ways as an engineer 😅  (So maybe worthwhile asking more representative target customers than me is what I'm admitting!)

Reply:
I'll add `READ_WRITE`, I can see how not having it is confusing... thank you!

Reply:
trying to think about what it means to read/write in the cases we have now - s3, ODP, kafka

s3 - It feels like a typical pattern is to read from one or multiple tables, clean and write to a new table in a single job
ODP - do we forsee folks reading the current value and then updating it? or would they read from a stream or s3 and then write to ODP?
kafka - I'm less sure here 😅

Reply:
Some may read and write to the same source (aka `dataset`) in the same step, example: reading and writing to a smoothed scores Hive table, or an ODP value.